nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchHttp.java

```diff
@@ -0,0 +1,330 @@
+            .build();
+
+    public static final PropertyDescriptor INDEX_OP = new PropertyDescriptor.Builder()
+            .name("Index Operation")
+            .description("The type of the operation used to index (index, update, upsert, delete)")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(
+                    AttributeExpression.ResultType.STRING, true))
+            .defaultValue("index")
+            .build();
+
+    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
+            .description("The preferred number of FlowFiles to put to the database in a single transaction")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("100")
+            .build();
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_RETRY);
+        return Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(ES_URL);
+        descriptors.add(PROP_SSL_CONTEXT_SERVICE);
+        descriptors.add(CONNECT_TIMEOUT);
+        descriptors.add(RESPONSE_TIMEOUT);
+        descriptors.add(ID_ATTRIBUTE);
+        descriptors.add(INDEX);
+        descriptors.add(TYPE);
+        descriptors.add(CHARSET);
+        descriptors.add(BATCH_SIZE);
+        descriptors.add(INDEX_OP);
+        return Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> problems = new ArrayList<>(super.customValidate(validationContext));
+        // Since Expression Language is allowed for index operation, we can't guarantee that we can catch
+        // all invalid configurations, but we should catch them as soon as we can. For example, if the
+        // Identifier Attribute property is empty, the Index Operation must evaluate to "index".
+        String idAttribute = validationContext.getProperty(ID_ATTRIBUTE).getValue();
+        String indexOp = validationContext.getProperty(INDEX_OP).getValue();
+
+        if(StringUtils.isEmpty(idAttribute)) {
+            if(StringUtils.isEmpty(indexOp)) {
+                problems.add(new ValidationResult.Builder()
+                        .valid(false)
+                        .subject(INDEX_OP.getDisplayName())
+                        .explanation("If Identifier Attribute is not set, Index Operation must evaluate to \"index\"")
```

Comment:
Unless if I'm misreading this seems to just check if both "INDEX_OP" and "ID_ATTRIBUTE" are empty. I believe INDEX_OP cannot be empty because it is required, has a default and is validated on the EL validator. 

Like you explain above, since you support EL in INDEX_OP you won't be able to fully validate that the ID_ATTRIBUTE is set when INDEX_OP != "index". That said, you can check if INDEX_OP is set explicitly as "update", "upsert" or "delete" and ID_ATTRIBUTE but you will still have to check in the onTrigger if the evaluated INDEX_OP is "index" if ID_ATTRIBUTE is not set.